src/main/java/edu/hm/hafner/coverage/parser/CloverParser.java

```diff
@@ -0,0 +1,234 @@
+                }
+            }
+        }
+        throw createEofException(fileName);
+    }
+
+    @CanIgnoreReturnValue
+    private FileNode readFile(final XMLEventReader reader, final PackageNode packageNode,
+            final StartElement fileElement) throws XMLStreamException {
+        String fileName = getValueOf(fileElement, NAME);
+        String className = fileName.substring(0, fileName.lastIndexOf("."));
+        String filePath = getValueOf(fileElement, PATH);
+        // TODO check if filePath is relative or absolute path (latter needs to remove everything except current workspace)
+        var fileNode = packageNode.findOrCreateFileNode(fileName, TreeString.valueOf(filePath));
+        var classNode = packageNode.findOrCreateClassNode(className);
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+            if (event.isStartElement()) {
+                var e = event.asStartElement();
+                if (METRICS.equals(e.getName())) {
+                    addBranchCoverage(classNode, e);
+                    addInstructionCoverage(classNode, e);
+                }
+                else if (LINE.equals(e.getName())) {
+                    int line = getIntegerValueOf(e, NUM);
+                    int count = getIntegerValueOf(e, COUNT);
+                    if (count > 0) {
+                        fileNode.addCounters(line, 1, 0);
+                    }
+                    else {
+                        fileNode.addCounters(line, 0, 1);
+                    }
+                }
+                else {
+                    new ParsingException(
+                            String.format("Unexpected element '%s' in <file> block in file '%s'", e.getName(),
+                                    fileName));
+                }
+            }
+            else if (event.isEndElement()) {
+                var endElement = event.asEndElement();
+                if (FILE.equals(endElement.getName())) {
+                    resolveLines(fileNode);
+                    return fileNode;
+                }
+            }
+        }
+        throw createEofException(fileName);
+    }
+
+    private void resolveLines(final FileNode fileNode) {
+        var val = createValue("LINE", fileNode.getCoveredLines().size(), fileNode.getMissedLines().size());
+        fileNode.addValue(val);
+        for (ClassNode c : fileNode.getAllClassNodes()) {
+            c.addValue(val);
+        }
+    }
+
+    private void addBranchCoverage(final Node node, final StartElement e) {// final int covered, final int total) {
+        int condTotal = getIntegerValueOf(e, CONDITIONALS);
+        int condCovered = getIntegerValueOf(e, COVERED_CONDITIONALS);
+        addCoverage(node, "BRANCH", condCovered, condTotal);
+    }
+
+    private void addInstructionCoverage(final Node node, final StartElement e) { //final int covered, final int total) {
+        int stmntsTotal = getIntegerValueOf(e, STATEMENTS);
```

Comment:
In Java we typically do not use shortened variable names, use `statementsTotal` and so on.